### DISCUSSION TITLE
"Sdk.props" was not found

### DISCUSSION THREAD

**Discussion by cpuwzd** [wixtoolset/issues#6307] [Questions] [answered]
I'm a newbie to Wix, but I've been programming for 59 years. I'm running the latest VS2019 version. I've run appveyor.cmd from the Tools folder. It generates errors when it gets to "CanBuildHeatProjectSdkStyle".  First, I see:

 D:\Repos\Wix\Tools>dotnet publish -c Release -o ...
... D:\Repos\Wix\Tools\build\Release\publish\WixToolset.Sdk\broken\net461\ ...
... -f net461 -r dne src\wix   || exit /b

then:

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
[xUnit.net 00:00:02.97]     WixToolsetTest.Sdk.MsbuildHeatFixture.CanBuildHeatProjectSdkStyle(buildSystem: MSBuild64, useToolsVersion: False) [FAIL]
  Failed WixToolsetTest.Sdk.MsbuildHeatFixture.CanBuildHeatProjectSdkStyle(buildSystem: MSBuild64, useToolsVersion: False) [1 s]
  Error Message:
   MSBuild failed unexpectedly. Output:
Microsoft (R) Build Engine version 15.9.21+g9802d43bc3 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

and:

Project "C:\Users\cpuwz\AppData\Local\Temp\qfzm52b4.r3a\HeatProjectSdkStyle\HeatProjectSdkStyle.wixproj" (1) is building "C:\Users\cpuwz\AppData\Local\Temp\qfzm52b4.r3a\SdkStyleCs\SdkStyleCs.csproj" (2) on node 1 (default targets).
C:\Users\cpuwz\AppData\Local\Temp\qfzm52b4.r3a\SdkStyleCs\SdkStyleCs.csproj : error MSB4019: The imported project "C:\Program Files\dotnet\sdk\3.1.100\Sdks\Microsoft.NET.Sdk\Sdk\Sdk.props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.
Done Building Project "C:\Users\cpuwz\AppData\Local\Temp\qfzm52b4.r3a\SdkStyleCs\SdkStyleCs.csproj" (default targets) -- FAILED.
Done Building Project "C:\Users\cpuwz\AppData\Local\Temp\qfzm52b4.r3a\HeatProjectSdkStyle\HeatProjectSdkStyle.wixproj" (default targets) -- FAILED.

Build FAILED.

This continues for a while until we get to:

CoreWixBuild:

Every target fails, and we eventually exit.

The folder "Broken" in one or more paths would seem to be a clue, but if that is the case, how can I install the current build? Is the current build simply broken? Is there a way to get the last build that was not broken? I assume that I could fork and clone a different branch or somehow use a label. Do I just have to wait for the broken build to be fixed?

Ultimately, I'd like to be able to debug through the source code and see if I can find someplace to help your efforts. I don't know how to convince MSBuild to use the code in my solution rather than the installed (if there is one) version. Is this documented anywhere? MSBuild has to find some number of targets defined in a targets file. I'm not sure how the targets files link back to the Wix implementation and how that can be diverted to a dormant debugging image. I suppose that JIT debugging might help. It would be fun to figure it out myself, but some pointers would be appreciated.

Sorry to be so long-winded. I'm old, and I've been long-winded for a long time.

Thanks for any help I might get.

Ron Martin

**Comment by rseanhall** [comment]
> D:\Repos\Wix\Tools\build\Release\publish\WixToolset.Sdk\broken\net461\  -f net461 -r dne src\wix || exit /b

This is to setup the environment for this test: https://github.com/wixtoolset/Tools/blob/d718243dfbb41f28aa8030939595b6493f4a78f0/src/test/WixToolsetTest.Sdk/MsbuildFixture.cs#L386

> C:\Users\cpuwz\AppData\Local\Temp\qfzm52b4.r3a\SdkStyleCs\SdkStyleCs.csproj : error MSB4019: The imported project "C:\Program Files\dotnet\sdk\3.1.100\Sdks\Microsoft.NET.Sdk\Sdk\Sdk.props" was not found. Confirm that the path in the declaration is correct, and that the file exists on disk.

I think this means you don't have the .NET Core 3.1 SDK installed on your machine?

> Is the current build simply broken?

If appveyor.cmd got this far (running the Sdk tests), then you've got the code successfully built on your machine. It's just the tests that are failing.

> Ultimately, I'd like to be able to debug through the source code and see if I can find someplace to help your efforts.

Are you running into a problem when using the toolset and trying to figure out how to fix it? Or are you just looking for something to work on?

**Reply by cpuwzd** [reply]
I've chased the .NET Core 3.1 SDK issue quite a bit. The folder structure is in place, but "C:\Program Files\dotnet\sdk\3.1.100\Sdks\Microsoft.NET.Sdk\Sdk\" is empty. I've tried installing both the 32-bit and the 64-bit versions of the SDK. Neither one complains. The .NET 3.1 runtime was probably installed first. Perhaps that is fooling the SDK installation into skipping some, or all, of the installation. Is there a way to uninstall the two versions of the SDK and the runtime, then reinstall the SDK?

I'm a little confused about whether the 32-bit, the 64-bit, or both versions should be installed. I'm running 64-bit Windows 10. Would the 32-bit version ever be used on this computer? I know that at least a major portion of VS 2019 is still 32-bit and I don't know if a 32-bit target signals the use of 32-bit tools. I've had some issues in the past with building a 64-bit target with satellite assemblies, where the satellite assemblies were built 32-bit and were incompatible with the primary assemblies.

Is the SDK installation logged or can it be logged? Which tool needs to have its logging turned on and where can I find the log files?

_(Added later)_
I'm checking out the Microsoft .NET Core SDK Uninstall Tool.
_(End of added later)_

Yes, I am trying to research a problem I was having with a build, but I discover that you already have somebody working on that problem, I would be willing to work on a different one.

I have some code running in .NET Framework that I am trying to port to .NET 5.0. I've stubbed out all the dependencies on the API's that are not yet released, so that I could get ahead on  the conversion. That's when I discovered that Wix 3 won't cut it and neither will Wix 4. So here I am.

I'm guessing that a lot of people are waiting for Wix to support .NET 5.0, so anything I can do to help the effort is fine with me. I've been retired for 13 years and I have lots of time that I spend playing games. I'd rather do something useful.

Ron

**Comment by rseanhall** [comment, marked as answer]
I'm going to be honest, I'm not interested in troubleshooting the missing SDK. All of the Sdk tests run using MSBuild 32-bit, MSBuild 64-bit, and MSBuild .NET Core. It's always worked for me using whatever Visual Studio installs by default so I don't know which architecture you need. I recommend uninstalling VS2019 and reinstalling it. I've attached my configuration in case you wanted to import it.

[20201222_vs2019182.vsconfig.txt](https://github.com/wixtoolset/issues/files/5731907/20201222_vs2019182.vsconfig.txt)

If you want something simple to start on you can look at https://github.com/wixtoolset/issues/issues/4753. I already wrote a failing test [here](https://github.com/rseanhall/Core/blob/85deb61f666f6817c1a137ace4d666c8ae2940fb/src/test/WixToolsetTest.CoreIntegration/RegistryFixture.cs#L115) so you just need to fix the code to get it passing.

In general, the current state of v4 supports .NET 5. We are currently working towards [preview.0](https://www.firegiant.com/blog/2020/6/24/wix-online-meeting-191-highlights/), so if you wanted to help with that you could look at the issues in the Needs Investigation state in the [preview.0 project](https://github.com/wixtoolset/issues/projects/1).

**Reply by rseanhall** [reply]
@cpuwzd

>"OneTimeWixBuildInitialization.proj" is mentioned in the documentation, e.g. at https://wixtoolset.org/development/building-wix/. I have seen it. I have run it. Now, starting over, I can't find any trace of it except in the version 3 repository. What happened to it?
>
>With a known bug in VS 2019, having referenced a preview version of the .NET 5.0 SDK, I can't publish anything, so I am dead in the water. I'm trying to figure out how you deal with building debug versions of local copies of the Wix repos. Two of the three platforms are disabled in Debug mode, yet running tests from VS 2019 in debug mode references these other two platforms. I've also found dependencies on Debug versions from code in Release versions. Do you publish Debug versions the same way you publish Release versions, and does "OneTimeWixBuildInitialization.proj" set these up so that they can be built and used locally without being signed?
>
> Would it help if I read all the issues? Have these procedural changes been hashed out there?
>
> I'm trying hard to help, but you're not making it very easy.

> As, suggested, I've moved this comment back to discussion #6307 on the wix-devs-mailing-list.
>
> As far as posting this message as a comment goes, I just now found the home page for discussions. I started from a Google search and it wasn't clear that I needed to get to the parent node and how to do so after I landed inside the topic. I think that I've got it now.

FYI, we don't get notifications when people make edits so if you would like a response please post a new comment.

You said you brought this discussion to the wix-devs mailing list but I have not seen it, so I'm moving the discussion back here so we can get you setup there. Did you try to send an email to the mailing list yet? You'll need to subscribe first by sending an email with the title of `subscribe` to wix-devs-subscribe@lists.wixtoolset.org.
